src/PowerPing/Ping.cs

```diff
@@ -110,12 +111,12 @@ public void Listen()
                 PowerPing.Display.ListenIntroMsg();
 
                 // Listening loop
-                while (true) {
+                while (!cancellationToken.IsCancellationRequested) {
                     byte[] buffer = new byte[4096]; // TODO: could cause overflow?
                     
                     // Recieve any incoming ICMPv4 packets
                     EndPoint remoteEndPoint = new IPEndPoint(IPAddress.Any, 0);
-                    int bytesRead = listeningSocket.ReceiveFrom(buffer, ref remoteEndPoint);
+                    int bytesRead = Task.Run(() => listeningSocket.ReceiveFrom(buffer, ref remoteEndPoint)).WaitForResult(cancellationToken);
```

Comment:
Sorry forgot to go back to this function in the review, again might be nice to have a comment explaining whats going on on here.